### PR TITLE
feat(policy): 1436 tech debt migrate subject mapping and condition set object queries to sqlc

### LINKED ISSUE #1436: Migrate Subject Mapping object queries to sqlc

### LINKED ISSUE #1437: Migrate Subject Condition Set object queries to sqlc

### PR DESCRIPTION
**Pull request by ryanulit**
fix #1436
fix #1437 

also implements:
- architecture decisions per [adr readme
](https://github.com/opentdf/platform/blob/main/service/policy/adr/0000-current-state.md)
- recent audit improvements

### CODE REVIEW

File: service/policy/db/query.sql [hunk 1]
```diff
@@ -630,4 +630,95 @@ VALUES ($1, $2);
...
+    JSON_BUILD_OBJECT(
+        'id', scs.id,
+        'metadata', JSON_STRIP_NULLS(JSON_BUILD_OBJECT('labels', scs.metadata->'labels', 'created_at', scs.created_at, 'updated_at', scs.updated_at)),
+        'subject_sets', scs.condition
+    ) AS subject_condition_set,
+    JSON_BUILD_OBJECT('id', av.id,'value', av.value,'active', av.active) AS attribute_value
+FROM subject_mappings sm
+LEFT JOIN attribute_values av ON sm.attribute_value_id = av.id
+LEFT JOIN attribute_definitions ad ON av.attribute_definition_id = ad.id
+LEFT JOIN attribute_namespaces ns ON ad.namespace_id = ns.id
```
Comment (jakedoublev) [inline service/policy/db/query.sql:687]: Do we need the definition and namespace here?

File: service/policy/db/query.sql [hunk 2]
```diff
@@ -630,4 +630,95 @@ VALUES ($1, $2);
...
+    JSON_BUILD_OBJECT(
+        'id', scs.id,
+        'metadata', JSON_STRIP_NULLS(JSON_BUILD_OBJECT('labels', scs.metadata -> 'labels', 'created_at', scs.created_at, 'updated_at', scs.updated_at)),
+        'subject_sets', scs.condition
+    ) AS subject_condition_set,
+    JSON_BUILD_OBJECT('id', av.id,'value', av.value,'active', av.active) AS attribute_value
+FROM subject_mappings sm
+LEFT JOIN attribute_values av ON sm.attribute_value_id = av.id
+LEFT JOIN attribute_definitions ad ON av.attribute_definition_id = ad.id
+LEFT JOIN attribute_namespaces ns ON ad.namespace_id = ns.id
```
Comment (jakedoublev) [inline service/policy/db/query.sql:705]: same question - are these necessary?

**Review by jakedoublev — APPROVED** [review]

File: service/policy/db/query.sql (hunk 1 above)
Comment (ryanulit) [reply service/policy/db/query.sql]: nope, removed

File: service/policy/db/query.sql (hunk 2 above)
Comment (ryanulit) [reply service/policy/db/query.sql]: removed as well

**Pull request merged**